kopf/reactor/registries.py

```diff
@@ -13,52 +13,96 @@
 """
 import abc
 import functools
+import logging
 from types import FunctionType, MethodType
-from typing import MutableMapping, NamedTuple, Text, Optional, Tuple, Callable, Mapping
+from typing import (Any, MutableMapping, Optional, Sequence, Collection, Iterable, Iterator,
+                    NamedTuple, Union, Tuple, List, Set, FrozenSet, Mapping, NewType, cast)
 
+from typing_extensions import Protocol
+
+from kopf.reactor import causation
+from kopf.structs import bodies
+from kopf.structs import dicts
+from kopf.structs import diffs
+from kopf.structs import patches
 from kopf.structs import resources as resources_
 
+# Strings are taken from the users, but then tainted as this type for stricter type-checking:
+# to prevent usage of some other strings (e.g. operator id) as the handlers ids.
+HandlerId = NewType('HandlerId', str)
+
+
+class HandlerFn(Protocol):
+    def __call__(self,
+                 *args: Any,
+                 type: str,
+                 event: Union[str, bodies.Event],  # FIXME: or str for cause-handlers.
```

Comment:
Yes, indeed. Removed. 

This will be fixed in the following 2 PRs with the terminology fixes (this field is renamed to `reason` there). Though, the backward-compatibility `event` kwargs/fields will be kept for indefinite time.